service/src/main/kotlin/io/provenance/digitalcurrency/consortium/extension/ProtoExtension.kt

```diff
@@ -14,16 +13,15 @@ fun newPaginationBuilder(offset: Int, limit: Int): PageRequest.Builder =
 
 fun Message.toAny(typeUrlPrefix: String = ""): Any = Any.pack(this, typeUrlPrefix)
 
-fun Iterable<Any>.toTxBody(memo: String? = null): TxBody =
+fun Iterable<Any>.toTxBody(timeoutHeight: Long): TxBody =
     TxBody.newBuilder()
         .addAllMessages(this)
-        .also { builder -> memo?.run { builder.memo = this } }
+        .setTimeoutHeight(timeoutHeight)
```

Comment:
im not too familiar with timeoutHeight - what is it for?

Reply:
if I understand it correctly, it means if the txn hasn't been included by a certain block height, don't 'commit' it. Sometimes we can receive a `timed out waiting for tx to be included in a block` error but the txn is ultimately included in a block. This helps mitigate that but now I'm wondering if I shouldn't store the timeout block and not retry in the async handler until that has been reached.

Reply:
i assume event stream ignores these errors right? and the tx expiration reaper picks this up and result is error?

i removed the height check from the retry resetter because if a transaction is in an error state it doesnt make sense to skip and wait for the block height to be reached before re-queuing (i ran into this when didnt provide enough hash for transaction)

Reply:
ok. When I did the event stream for usdf omni I added logic to skip errors but I'm not sure they even make it that far. So yes the expired reaper would pick it up. 

I'm still fuzzy on the timeout height in action so good to know that you've seen it. I'll propagate the change to omni usdf.

Reply:
yeah i'm not sure how to test a broadcast tx height timeout - i would assume that it would just error after a certain blockheight and the reaper would indicate as such